ios/Classes/PMPlugin.m

```diff
@@ -16,22 +16,38 @@
 @implementation PMPlugin {
     BOOL ignoreCheckPermission;
     NSObject <FlutterPluginRegistrar> *privateRegistrar;
+    FlutterMethodChannel* channel;
+    NSMutableArray* progressHandlerDisposeCallbacks;
```

Comment:
Can we have a better structure here? Setting up an array for fallbacks seems overhead for the plugin instance.

Reply:
What about the classic delegate pattern? 
Creating a delegate that `PMPlugin` will implement and then passing it to the progress handler?